packages/shaderLab/src/parser/tokens/keyword.ts

```diff
@@ -0,0 +1,43 @@
+import { ValueString } from "./value";
+import { createKeywordToken } from "./utils";
+
+export const Shader = createKeywordToken("Shader", { longer_alt: ValueString });
+export const EditorProperties = createKeywordToken("EditorProperties");
+export const SubShader = createKeywordToken("SubShader");
+export const Pass = createKeywordToken("Pass");
+export const Tags = createKeywordToken("Tags");
+
+export const BlendState = createKeywordToken("BlendState");
+export const DepthState = createKeywordToken("DepthState");
+export const StencilState = createKeywordToken("StencilState");
+export const RasterState = createKeywordToken("RasterState");
+
+export const Enabled = createKeywordToken("Enabled");
+export const SrcColorBlendFactor = createKeywordToken("SrcColorBlendFactor");
+export const DestColorBlendFactor = createKeywordToken("DestColorBlendFactor");
```

Comment:
Where to consume these variables，`RenderState` is in `Material` instead of `Shader` now

Reply:
> Where to consume these variables，`RenderState` is in `Material` instead of `Shader` now

1. The keywords mentioned above is for shader DSL parser only, and invisible to engine user
2. yes, the renderstate keywords should be present in `subshader` or `pass` block

Reply:
All shader grammer rules is declared in `src/parser/rules`

Reply:
yes，so you should put these variables from `Material` to `Shader` to enable these function

Reply:
yes，so you should put these variables from Material to Shader to enable these function